ui/src/wizard/ResourceGroupStep.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { ResourceGroup } from 'azure-arm-resource/lib/resource/models';
+import { OutputChannel } from 'vscode';
+import { IAzureNamingRules, IAzureQuickPickItem, IAzureQuickPickOptions, IAzureUserInput, IResourceGroupWizardContext } from '../../index';
+import { localize } from '../localize';
+import { uiUtils } from '../utils/uiUtils';
+import { AzureWizardStep } from './AzureWizardStep';
+
+export const resourceGroupNamingRules: IAzureNamingRules = {
+    minLength: 1,
+    maxLength: 90,
+    invalidCharsRegExp: /[^a-zA-Z0-9\.\_\-\(\)]/
+};
+
+export class ResourceGroupStep<T extends IResourceGroupWizardContext> extends AzureWizardStep<T> {
+    private _newName: string;
+
+    public static async getResouceGroups<T extends IResourceGroupWizardContext>(wizardContext: T): Promise<ResourceGroup[]> {
+        if (wizardContext.resourceGroupsTask === undefined) {
+            // tslint:disable-next-line:no-non-null-assertion
+            const client: ResourceManagementClient = new ResourceManagementClient(wizardContext.credentials, wizardContext.subscription.subscriptionId!);
+            wizardContext.resourceGroupsTask = uiUtils.listAll(client.resourceGroups, client.resourceGroups.list());
+        }
+
+        return await wizardContext.resourceGroupsTask;
+    }
+
+    public static async isNameAvailable<T extends IResourceGroupWizardContext>(wizardContext: T, name: string): Promise<boolean> {
```

Comment:
The resourceManagementClient has a function called checkExistence that does most of this name validation.   
https://github.com/Azure/azure-sdk-for-node/blob/1640f6952c3510547c3e02aac1c663d0b7354736/lib/services/resourceManagement/lib/resource/operations/resourceGroups.js#L40-L80

Reply:
I didn't know that existed, but again I'm not a huge fan of those error messages. I think they're worded weirdly, but more importantly I hate error messages that include regular expressions. I just don't think regular expressions are easily readable.

For those reasons, I think I'll keep it as-is for this PR. But like you said it would be easy to change later and I'm open to the idea

Reply:
Perhaps we should complain to the SDK and give them suggestions. I agree they stink.